pkg/agent/externalnode/external_node_controller.go

```diff
@@ -552,12 +552,23 @@ func (c *ExternalNodeController) updateOVSPortsData(interfaceConfig *interfacest
 func (c *ExternalNodeController) removeOVSPortsAndFlows(interfaceConfig *interfacestore.InterfaceConfig) error {
 	portUUID := interfaceConfig.PortUUID
 	portName := interfaceConfig.InterfaceName
+	hostIFName := interfaceConfig.InterfaceName
+	uplinkIfName := util.GenerateUplinkInterfaceName(portName)
+
+	// This is for issue #5111 (https://github.com/antrea-io/antrea/issues/5111), which may happen if an error occurs
+	// when moving the configuration back from host internal interface to uplink. This logic is run in the second
+	// try after the error is returned, at this time the host internal interface is already deleted, and the uplink's
+	// name is recovered. So the ips and routes in "adapterConfig" are actually read from the uplink and no need to
```

Comment:
I may miss some thing here. If the error occurs, the uplink's name is recovered.  But ip and routes can still be wrong, right? Our current logic can not handle this case and make it right later?

Reply:
For your mentioned case, we can not fix it in this change, and it may also break agent without this change.

The motivation for this change is issue #5111 , in which ip address/routes configuration are failed because dhclient is modifying the interface in the meanwhile. Then antrea-agent may fail in the first try, and it still blocks at the check on the existence host internal interface (antrea thought the interface was supposed to not exist because it was removed from OVS and the uplink was not renamed yet), but the fact is the uplink is already renamed. As for the ip address/routes are actually configured back by dhclient.

Reply:
Thanks for the background details. It is clear to me now.